### DISCUSSION TITLE
Changing font size in all OS windows at once

### DISCUSSION THREAD

**Discussion by stinovlas** [kovidgoyal/kitty#7393] [Q&A] [answered]
Hi! I recently migrated to `kitty` from `alacritty`. I have to say `kitty` is amazing and I'm still going through a heap of useful features it provides out of the box. One thing I've been struggling with is changing font size in all OS windows at once. This is very useful feature for me, because I carry my laptop around and attach it to differently sized monitors with different resolutions.

## What is the problem
The documentation says that `map kitty_mod+plus   change_font_size all +2.0` should affect all OS windows, but curiously, it didn't work for me. I use `i3` and I had the following in my `i3` config:

```
bindsym $mod+Shift+Return exec kitty
```

It turned out, when you execute kitty this way, it spawns several `kitty` instances. And `change_font_size all` only affects all the windows in single `kitty` instance.

## My solution
First I wanted to utilize remote control, but then I stumbled into useful command line option `--single-instance`. If you use it, `kitty` will attach new windows to an existing instance instead of creating a new one. So I changed my `i3` config to this and everything works like a charm:

```
bindsym $mod+Shift+Return exec kitty --single-instance
```

Note: there's also `--instance-group` CLI option, if you want to group windows into several separate instances. I don't currently have a use for that.

*Note: I understand this is a weird Q&A, because I already found the answer, but I spent some time with it and I thought someone else might find it useful. Furthermore, there's still an open question of affecting the font size over all running `kitty` instances (remote control?) if anyone has use for that.*

**Comment by kovidgoyal** [comment, marked as answer]
Just for completeness, you can also update the value in your kitty.conf
file and have all kitty processes reload their config wit something like

killall -USR1 kitty

**Reply by stinovlas** [reply]
Thank you, this is useful! I can imagine having a small config file included in the main `kitty.conf` which I would edit with in-place `sed` and reload the config as you suggested. Not necessary for the simultaneous font size change, but maybe I'll find another user for it.
